File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.* ;
class Solution {
    public int solution(int[] priorities, int location) {
     
        Obj[] obj = new Obj[priorities.length] ; 
        for(int i = 0 ; i<priorities.length ; i++){
            obj[i] = new Obj(i ,  priorities[i]); 
        }
        PriorityQueue<Obj> que = new PriorityQueue<>((a, b) -> b.num - a.num); 
        ArrayDeque<Obj> deque = new ArrayDeque<>() ; 
        for(Obj o : obj){
            que.add(o) ;
            deque.add(o); 
        }
    
        List<Integer> list = new ArrayList<>() ; 
        while(!que.isEmpty()){
            Obj o1 = que.peek();
            Obj o2 = deque.poll();
            if(o1.num == o2.num){
                list.add(o2.index)  ;
                 que.poll() ;
            }
            else{
                deque.add(o2); 
            }
          

        }      
        int answer = 0 ; 
        for(int i = 0  ;i<list.size() ; i++){
            if(location == list.get(i)){
                answer = i+1 ;
                break;
            }
        }
       
        return answer ;
    }     
    static class Obj {
        
        int index ; 
        int num ; 
        
        Obj(int index , int num){
            this.index = index ; 
            this.num = num ; 
        }
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    var answer = 0;
    const newArr = priorities.map((item , index) => {
        return {
            index , 
            item 
        }
    });  
    priorities.sort((a,b) => b-a) ;
    let result = [] ; 
    let num = 0 ; 
  
    while(priorities.length != 0){
        let p = priorities.shift(); 
        let n = newArr.shift() ; 
        if(p === n.item){
            
            result.push(n.index); 
            
        }
        else if(n.item !== p){
            newArr.push(n) ;
            priorities.unshift(p) ; 
                   
        }                         
      
    } 
    

      
    for(let i = 0 ; i<result.length ; i++){
        if(result[i] === location){
            return i+1 ; 
        }
    }
    return 1;
}